src/ReflectionParameter.php

```diff
@@ -89,7 +89,7 @@ public function __construct(
         $this->parameterIndex    = $parameterIndex;
         $this->declaringFunction = $declaringFunction;
 
-        if ($this->isDefaultValueAvailable()) {
+        if ($this->isDefaultValueSet()) {
```

Comment:
Why this is changed?

Reply:
Some built in functions and methods have optional parameters **without** a default value, which provides some interesting introspection values. This is part of the handling of this case.

Reply:
I did some thinking on this. I think I'm able to do the same thing with a more minor change to `ReflectionParameter`. Let me take another crack at this. 

Everything I said above is still valid, but I think I have a more subtle solution with the same behavior.

Reply:
> optional parameter with no default value

Then what is it? You haven't provided any example PHP code.

> Everywhere else in this PR, if we don't have source code for a function/method/class/etc.

OK. Since this PR doesn't sound like something that can be finalized/merged this year, then I thought at least `optional parameter with no default value` (whatever this is) could be merged as separately PR. I was wrong apparently. No actions needed from you here.

Reply:
I apologize for not being more responsive lately. I have had far less time to devote to this project than I anticipated, but I am still hopeful to have this merged in, within 2 to 4 weeks with everything else going on.

As far as an example of a builtin function like this, take `preg_match()` as an example:

```php
<?php

$preg_matchRef = new ReflectionFunction('preg_match');
$params = $preg_matchRef->getParameters();
var_export(['getName()' => $params[2]->getName(),
'isOptional()' => $params[2]->isOptional(),
'isDefaultValueAvailable()' => $params[2]->isDefaultValueAvailable()]);
print "\n\n";
```

The `&$matches` parameter (as it appears in the documentation, although it's called `$subpatterns` internally, is optional, but has no default value. To my knowledge, there is no way to represent this in PHP source code.

Reply:
> I apologize for not being more responsive lately.

Not to worry. We're not on a schedule here 😉 

> To my knowledge, there is no way to represent this in PHP source code.

Haven't thought about this scenario. I guess this is what you get, when attempt to reflect built-in functions :(